lib/vfvalue.h

```diff
@@ -262,6 +268,53 @@ namespace ValueFlow
         /** The value bound  */
         Bound bound = Bound::Point;
 
+        /** value relies on safe checking */
+        // cppcheck-suppress premium-misra-cpp-2023-12.2.1
+        bool safe : 1;
+
+        /** Conditional value */
+        bool conditional : 1;
+
+        /** Value is is from an expanded macro */
+        bool macro : 1;
+
+        /** Is this value passed as default parameter to the function? */
+        bool defaultArg : 1;
+
+        long long : 4; // padding
+
+        /** kind of moved  */
+        enum class MoveKind : std::uint8_t { NonMovedVariable, MovedVariable, ForwardedVariable } moveKind = MoveKind::NonMovedVariable;
+
+        enum class LifetimeScope : std::uint8_t { Local, Argument, SubFunction, ThisPointer, ThisValue } lifetimeScope = LifetimeScope::Local;
+
+        enum class LifetimeKind : std::uint8_t {
+            // Pointer points to a member of lifetime
+            Object,
+            // A member of object points to the lifetime
+            SubObject,
+            // Lambda has captured lifetime(similar to SubObject)
+            Lambda,
+            // Iterator points to the lifetime of a container(similar to Object)
+            Iterator,
+            // A pointer that holds the address of the lifetime
+            Address
+        } lifetimeKind = LifetimeKind::Object;
+
+        /** How known is this value */
+        enum class ValueKind : std::uint8_t {
+            /** This value is possible, other unlisted values may also be possible */
+            Possible,
+            /** Only listed values are possible */
+            Known,
+            /** Inconclusive */
+            Inconclusive,
+            /** Listed values are impossible */
+            Impossible
+        } valueKind = ValueKind::Possible;
+
+        std::int8_t indirect{}; // TODO: can we reduce the size?
```

Comment:
An old wisdom is that it is a good idea to order members by size.
See i.e.: http://katecpp.github.io/struct-members-order/
I have seen good results in a embedded project.
I guess it's pretty platform-dependent and would make no difference on some platforms.

would it be a good idea to use "pragma packed" on this struct?